src/libaktualizr/uptane/tuf.h

```diff
@@ -258,42 +260,17 @@ class Target {
    */
   bool IsOstree() const;
 
-  bool operator==(const Target &t2) const {
-    // if (type_ != t2.type_) return false; // Director doesn't include targetFormat
-    if (filename_ != t2.filename_) {
-      return false;
-    }
-    if (length_ != t2.length_) {
-      return false;
-    }
-
-    // requirements:
-    // - all hashes of the same type should match
-    // - at least one pair of hashes should match
-    bool oneMatchingHash = false;
-    for (const Hash &hash : hashes_) {
-      for (const Hash &hash2 : t2.hashes_) {
-        if (hash.type() == hash2.type() && !(hash == hash2)) {
-          return false;
-        }
-        if (hash == hash2) {
-          oneMatchingHash = true;
-        }
-      }
-    }
-    return oneMatchingHash;
-  }
-
+  bool operator==(const Target &t2) const;
   Json::Value toDebugJson() const;
   friend std::ostream &operator<<(std::ostream &os, const Target &t);
 
  private:
   bool valid{true};
   std::string filename_;
   std::string type_;
-  std::map<EcuSerial, HardwareIdentifier> ecus_;
+  std::map<EcuSerial, HardwareIdentifier> ecus_;  // Director only
```

Comment:
That's what I meant in one of the earlier comment, a field is used just for a specific type of Target instances.